programs/drift/src/instructions/admin.rs

```diff
@@ -4438,6 +4441,169 @@ pub fn handle_update_protected_maker_mode_config(
+            .get_token_amount(&spot_market)?
+            .cast::<u64>()?
+            .min(amount)
+    } else {
+        amount
+    };
+
+    let total_deposits_after = user.total_deposits;
+    let total_withdraws_after = user.total_withdraws;
+
+    let spot_position = &mut user.spot_positions[position_index];
+    controller::spot_position::update_spot_balances_and_cumulative_deposits(
+        amount as u128,
+        &SpotBalanceType::Deposit,
+        &mut spot_market,
+        spot_position,
+        false,
+        None,
+    )?;
+
+    let token_amount = spot_position.get_token_amount(&spot_market)?;
+    if token_amount == 0 {
+        validate!(
+            spot_position.scaled_balance == 0,
+            ErrorCode::InvalidSpotPosition,
+            "deposit left user with invalid position. scaled balance = {} token amount = {}",
+            spot_position.scaled_balance,
+            token_amount
+        )?;
+    }
+
+    if spot_position.balance_type == SpotBalanceType::Deposit && spot_position.scaled_balance > 0 {
+        validate!(
+            matches!(spot_market.status, MarketStatus::Active),
+            ErrorCode::MarketActionPaused,
+            "spot_market not active",
+        )?;
+    }
+
+    drop(spot_market);
+    if user.is_being_liquidated() {
+        // try to update liquidation status if user is was already being liq'd
+        let is_being_liquidated = is_user_being_liquidated(
+            user,
+            &perp_market_map,
+            &spot_market_map,
+            &mut oracle_map,
+            state.liquidation_margin_buffer_ratio,
+        )?;
+
+        if !is_being_liquidated {
+            user.exit_liquidation();
+        }
+    }
+
+    user.update_last_active_slot(slot);
+
+    let spot_market = &mut spot_market_map.get_ref_mut(&market_index)?;
+
+    controller::token::receive(
+        &ctx.accounts.token_program,
+        &ctx.accounts.admin_token_account,
+        &ctx.accounts.spot_market_vault,
+        &ctx.accounts.admin,
+        amount,
+        &mint,
+    )?;
+    ctx.accounts.spot_market_vault.reload()?;
+
+    let deposit_record_id = get_then_update_id!(spot_market, next_deposit_record_id);
+    let oracle_price = oracle_price_data.price;
+    let deposit_record = DepositRecord {
+        ts: now,
+        deposit_record_id,
+        user_authority: user.authority,
+        user: user_key,
+        direction: DepositDirection::Deposit,
+        amount,
+        oracle_price,
+        market_deposit_balance: spot_market.deposit_balance,
+        market_withdraw_balance: spot_market.borrow_balance,
+        market_cumulative_deposit_interest: spot_market.cumulative_deposit_interest,
+        market_cumulative_borrow_interest: spot_market.cumulative_borrow_interest,
+        total_deposits_after,
+        total_withdraws_after,
+        market_index,
+        explanation: DepositExplanation::Reward,
+        transfer_user: Some(ctx.accounts.admin.key()),
```

Comment:
might wanna check if this will confuse ui?

Reply:
i think we can make this None, since explanation is already Reward

Reply:
unless we want to avoid creating a new `Reward` enum and do something confusing like 'transfer_user == admin` to signify a reward.